lib/date_time/date_time.c

```diff
@@ -99,12 +108,88 @@ static void date_time_schedule_update(void)
 	}
 }
 
+static int current_time_check(void)
+{
+	if (time_aux.last_date_time_update == 0 ||
+	    time_aux.date_time_utc == 0) {
+		LOG_DBG("Date time never set");
+		return -ENODATA;
+	}
+
+	if ((k_uptime_get() - time_aux.last_date_time_update) >=
+	    CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS * MSEC_PER_SEC) {
+
+		LOG_DBG("Current date time too old");
+		return -ENODATA;
+	}
+
+	return 0;
+}
+
+static void date_time_store(int64_t curr_time_ms)
+{
+	struct timespec tp = { 0 };
+	struct tm ltm = { 0 };
+	int ret;
+
+	tp.tv_sec = curr_time_ms / 1000;
+	tp.tv_nsec = (curr_time_ms % 1000) * 1000000;
+
+	ret = clock_settime(CLOCK_REALTIME, &tp);
+	if (ret != 0) {
+		LOG_ERR("Could not set date %d", ret);
+		return;
+	}
+	gmtime_r(&tp.tv_sec, &ltm);
+	LOG_DBG("System time updated: %04u-%02u-%02u %02u:%02u:%02u",
+		ltm.tm_year + 1900, ltm.tm_mon + 1, ltm.tm_mday,
+		ltm.tm_hour, ltm.tm_min, ltm.tm_sec);
+
+#if defined(CONFIG_DATE_TIME_MODEM)
+	/* Set modem time if it's not valid */
+	/* TODO: KCONFIG ON WHETHER MODEM TIME IS UPDATED? */
```

Comment:
Should we have this flag? Given how simple the implementation has become, I'd not add it.
I'm removing this TODO from the code.

Reply:
I think the current implementation should be safe and thus there should no need for a flag.